components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java

```diff
@@ -1036,6 +1024,208 @@ public String getLastUpdatedTimeOfComment(String commentId) throws APIMgtDAOExce
         return EntityDAO.getLastUpdatedTimeOfResourceByUUID(AM_API_COMMENTS_TABLE_NAME, commentId);
     }
 
+    @Override
+    public void addRating(String apiId, Rating rating) throws APIMgtDAOException {
+        final String addRatingQuery =
+                "INSERT INTO AM_API_RATINGS (UUID, API_ID, RATING, USER_IDENTIFIER, " +
+                        "CREATED_BY, CREATED_TIME, UPDATED_BY, LAST_UPDATED_TIME" + ") VALUES (?,?,?,?,?,?,?,?)";
+        try (Connection connection = DAOUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(addRatingQuery)) {
+            try {
+                connection.setAutoCommit(false);
+                statement.setString(1, rating.getUuid());
+                statement.setString(2, apiId);
+                statement.setInt(3, rating.getRating());
+                statement.setString(4, rating.getUsername());
+                statement.setString(5, rating.getCreatedUser());
+                statement.setTimestamp(6, Timestamp.valueOf(LocalDateTime.now()));
+                statement.setString(7, rating.getLastUpdatedUser());
+                statement.setTimestamp(8, Timestamp.valueOf(LocalDateTime.now()));
+                statement.execute();
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                String errorMessage =
+                        "Error while adding rating for api id: " + apiId;
+                log.error(errorMessage, e);
+                throw new APIMgtDAOException(e);
+            }
+        } catch (SQLException e) {
+            log.error("Error while creating database connection/prepared-statement", e);
+            throw new APIMgtDAOException(e);
+        }
+    }
+
+    @Override
+    public Rating getRatingByUUID(String apiId, String ratingId) throws APIMgtDAOException {
+        final String query = "SELECT UUID, API_ID, RATING, USER_IDENTIFIER, " +
+                "CREATED_BY, CREATED_TIME, UPDATED_BY, LAST_UPDATED_TIME "
+                + "FROM AM_API_RATINGS WHERE UUID = ? AND API_ID = ?";
+        try (Connection connection = DAOUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            try {
+                statement.setString(1, ratingId);
```

Comment:
Shouldn't we set auto commit to false and commit to avoid datasource default behavior?